### ISSUE TITLE
[Tab Title] Provide high contrast support to `closable` button

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#6994]
### Summary

The `tab-title` text has sufficient contrast in high contrast mode: 
![Capture](https://github.com/Esri/calcite-components/assets/5023024/bdbde7b2-5726-4f80-83cb-98be5a68abce)

However, the new `closable` tab button, added from #2620, only provides some context on focus:
![Capture2](https://github.com/Esri/calcite-components/assets/5023024/25feb8b4-d775-435c-8c1b-7a03f7af130f)

### Actual Behavior

The `tab title` closable button does not provide sufficient context (see image above in the "Summary" section).

### Expected Behavior

Additional context should be provided to the the `closable` button using the high contrast CSS media feature, [`forced-colors`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors).

![expected high contrast behavior](https://github.com/Esri/calcite-components/assets/5023024/54a1742e-1dd8-4e19-b445-59c9965c0af2)

### Reproduction Sample

https://codepen.io/geospatialem/pen/wvYRRbN

### Reproduction Steps

1. Open the [following Codepen](https://codepen.io/geospatialem/pen/wvYRRbN)
2. Turn on the OS' high contrast (Windows)
3. Select the first item
4. Observe the contrast ratio

### Reproduction Version

1.4.0

### Relevant Info

High contrast was determined in the closable button as part of #2620

### Priority impact

p1 - need for current milestone

### Esri team

Calcite (dev)

**Comment by mpriour** [comment]
@geospatialem I can take care of this next week if you like.

**Comment by geospatialem** [comment]
@mpriour Awesome! Adding you in for assignment, thanks much, Matt!

**Comment by geospatialem** [comment]
Verified on Windows with high contrast enabled and `1.5.0-next.17`.

![image](https://github.com/Esri/calcite-design-system/assets/5023024/765ab41a-696a-4694-95d1-59b2e52530b2)
